### DISCUSSION TITLE
How to run teler as a systemd service?

### DISCUSSION THREAD

**Discussion by 0xjmux** [teler-sh/teler#274] [Q&A] [answered]
I'm trying to run teler against an nginx access log, like the below from the docs:
```
$ tail -f /var/log/nginx/access.log | teler -c /path/to/config/teler.yaml
```
However, all the resources I've found on the wiki/site have you run it in an active terminal session, or only analyze logs at a specific point in time. I'd like teler to analyze logs in real time, as they come in, and export it's own log in real time so it can be picked up by my SIEM system - which is configured to monitor teler's log file `/var/log/teler.log`. 

My many attempts to convert `teler` into a systemd service have failed - you can't have pipes in service commands, for one. I won't go through all the things I've tried, but I figure there has to be a way to run teler as a service that can be set up to run in the background, instead of live in the foreground taking up a tty. 

Is there a way to do this, and if so, how? Thanks.

**Comment by dwisiswant0** [comment, marked as answer]
For now, teler does not support running in the background. It looks like it's possible to run the teler placed on _systemd_, for example in the resource I found [here](https://www-linuxsec-org.translate.goog/2020/12/konfigurasi-teler-real-time-http.html?_x_tr_sl=id&_x_tr_tl=en&_x_tr_hl=en&_x_tr_pto=wapp) ([original](https://linuxsec.org/2020/12/konfigurasi-teler-real-time-http.html)).

But this has been our consideration to be able to do this on https://github.com/kitabisa/teler/issues/276 (dependent https://github.com/kitabisa/teler/issues/277).

**Comment by dwisiswant0** [comment]
Another straight tutorial on: 

Here's a simple example of daemonizing with automatic reloading on config change:

https://gist.github.com/asheroto/ff192448bb27b27e7c2d7cefdc24f308

_Originally posted by @asheroto in https://github.com/kitabisa/teler/issues/276#issuecomment-1924388271_
